Source/WebCore/rendering/StyledMarkedText.cpp

```diff
@@ -136,18 +210,38 @@ Vector<StyledMarkedText> StyledMarkedText::subdivideAndResolve(const Vector<Mark
         return { styledMarkedText };
     }
 
-    auto markedTexts = MarkedText::subdivide(textsToSubdivide);
+    auto markedTexts = MarkedText::subdivide(textsToSubdivide, OverlapStrategy::None);
     ASSERT(!markedTexts.isEmpty());
     if (UNLIKELY(markedTexts.isEmpty()))
         return { };
 
+    bool isHighlights = false;
```

Comment:
I would name this `containsHighlight`
```suggestion
    bool containsHighlight = false;
```